src/main.ts

```diff
@@ -62,13 +62,32 @@ export const loop = ErrorMapper.wrapLoop(() => {
 
         // Attempt to spawn a screep!
         let spawn = spawns[0];
-        spawn.spawnCreep([WORK, CARRY, MOVE], 'Worker1');
+        spawn.spawnCreep([WORK, MOVE, CARRY, MOVE], 'Worker1');
 
       } else {
         // It wasn't greater than 0... meaning we have no spawns.
         // This room can't create creeps.
       }
 
+        // get a reference to our creep
+
+        var mycreep = Game.creeps["Worker1"];
+        mycreep.say("I live!");
+
+        // if creep has no energy, go to the energy source and harvest some
+
+        if(mycreep.store[RESOURCE_ENERGY] == 0) {
+            var source = Game.getObjectById("f0a3a596141a8d65db841570");
```

Comment:
We will want to FIND sources in the room and use the id found. This will allow us to change rooms.
`const sources = room.find(FIND_SOURCES);`

Things we will also want to do is:
1. Find sources
2. Verify there are more than zero.
3. Use the first one to start with (sources[0]).
4. Then travel to, harvest, and return.

Reply:
Feels like this would be easier if you just did it for me

Reply:
Bumping this up; still haven't seen your code updates @smbader

Reply:
This suggestion is outdated. Pushed a possible solution.